funcx_endpoint/funcx_endpoint/endpoint/register_endpoint.py

```diff
@@ -75,3 +82,41 @@ def register_endpoint(
         )
 
     return reg_info["task_queue_info"], reg_info["result_queue_info"]
+
+
+def _get_endpoint_metadata(config: Config) -> dict:
+    metadata: t.Dict[str, t.Any] = dict()
+
+    metadata["version"] = funcx_endpoint.__version__
+
+    # let the web service get the client's IP address
+    metadata["ip_address"] = None
+    metadata["hostname"] = socket.gethostname()
+
+    # should be more accurate than `getpass.getuser()` in non-login situations
+    metadata["local_user"] = pwd.getpwuid(os.getuid()).pw_name
+
+    metadata["config"] = {
+        "executors": [
+            {
+                **_public_members(executor),
+                "provider": _public_members(executor.provider),
+                "launcher": _public_members(executor.provider),
+                "strategy": _public_members(executor.strategy),
+            }
+            for executor in config.executors
+            if isinstance(executor, HighThroughputExecutor)
+        ]
+    }
+
```

Comment:
Is there any value in spot-checking a couple of these in the eventual request?  Unless I'm missing something, it appears that the only test changes are for addition of the `default_config`.